src/adapters/input/http/controllers/product.ts

```diff
@@ -1,90 +1,106 @@
+        name: req.body.name,
+        description: req.body.description,
+        price: req.body.price,
+        image: req.file?.buffer,
+      };
+
+      const serviceRes = await this.productService.create(product);
 
       if (serviceRes.errorMessage !== undefined) {
-        return res.status(500).send(serviceRes)
+        return res.status(500).send(serviceRes);
       }
 
       if (!serviceRes.created) {
-        return res.status(400).json(serviceRes)
+        return res.status(400).json(serviceRes);
       }
 
-      return res.sendStatus(201)
-      } catch (error) {
-        console.log(error)
-        if (error instanceof Error){
-          return res.sendStatus(500)
-        }
-        throw(error)
+      return res.sendStatus(201);
+    } catch (error) {
+      if (error instanceof Error) {
+        return res.sendStatus(500);
       }
+      throw error;
     }
+  }
+
+  async editProduct(req: Request, res: Response): Promise<any> {
+    try {
+      const id = Number(req.params.id);
+      const product: Partial<Product> = req.body;
 
-    async deleteProduct(req: Request, res: Response): Promise<Response>  {
-      try {
-        const serviceRes = await this.productService.delete(Number(req.params.id))
-
-        if (serviceRes.errorMessage !== undefined){
-          return res.status(500).send(serviceRes)
-        }
-
-        if (!serviceRes.isValid) {
-          return res.sendStatus(400)
-        }
-
-        if (!serviceRes.wasFound){
-          return res.sendStatus(404)
-        }
-
-        return res.sendStatus(200)
-      } catch (error) {
-        console.log(error)
-        if (error instanceof Error){
-          return res.sendStatus(500)
-        }
-        throw(error)
+      if (req.file) {
+        product.image = req.file.buffer;
       }
+
+      const productResponse = await this.productService.edit(id, product);
+
+      return res.status(productResponse.status).send({ message: productResponse.message });
+    } catch (error) {
+      if (error instanceof Error) {
+        return res.sendStatus(500);
+      }
+      throw error;
     }
+  }
+
+  async deleteProduct(req: Request, res: Response): Promise<Response> {
+    try {
+      const serviceRes = await this.productService.delete(Number(req.params.id));
 
-    async listProductsByCategory(req: Request, res: Response): Promise<Response>  {
-      try {
-        const serviceRes = await this.productService.listByCategory(Number(req.params.id))
-
-        if (!serviceRes.isValid) {
-          return res.sendStatus(400)
-        }
-
-        return res.status(200).json(serviceRes.products)
-      } catch (error) {
-        console.log(error)
-        if (error instanceof Error){
-          return res.sendStatus(500)
-        }
-        throw(error)
+      if (serviceRes.errorMessage !== undefined) {
+        return res.status(500).send(serviceRes);
       }
+
+      if (!serviceRes.isValid) {
+        return res.sendStatus(400);
```

Comment:
I think it would be nice to add a message with the reason why we return a bad request. WDYT?

Reply:
Good idea. I'll do it